rpc/src/v1/traits/eth.rs

```diff
@@ -238,3 +385,386 @@ pub trait EthSigning: Sized + Send + Sync + 'static {
+		})
+	}
+
+	/// Returns block with given number.
+	fn block_by_number(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params::<(BlockNumber, bool)>(params).and_then(|(num, include_txs)| {
+			Eth::block_by_number(self, num, include_txs).map(to_value)
+		})
+	}
+
+	/// Returns the number of transactions sent from given address at given time (block number).
+	fn transaction_count(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params_default_second::<RpcH160>(params).and_then(|(address, block_number)| {
+			Eth::transaction_count(self, &address.into(), block_number)
+				.map(RpcU256::from).map(to_value)
+		})
+	}
+
+	/// Returns the number of transactions in a block with given hash.
+	fn block_transaction_count_by_hash(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params::<(RpcH256,)>(params).and_then(|(hash,)| {
+			Eth::block_transaction_count_by_hash(self, &hash.into())
+				.map(|x| x.map(RpcU256::from))
+				.map(to_value)
+		})
+	}
+
+	/// Returns the number of transactions in a block with given block number.
+	fn block_transaction_count_by_number(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params(params).and_then(|(block_number,)| {
+			Eth::block_transaction_count_by_number(self, block_number)
+				.map(|x| x.map(RpcU256::from))
+				.map(to_value)
+		})
+	}
+
+	/// Returns the number of uncles in a block with given hash.
+	fn block_uncles_count_by_hash(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params::<(RpcH256,)>(params).and_then(|(hash,)| {
+			Eth::block_uncles_count_by_hash(self, &hash.into())
+				.map(|x| x.map(RpcU256::from))
+				.map(to_value)
+		})
+	}
+
+	/// Returns the number of uncles in a block with given block number.
+	fn block_uncles_count_by_number(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params(params).and_then(|(block_number,)| {
+			Eth::block_uncles_count_by_number(self, block_number)
+				.map(|x| x.map(RpcU256::from))
+				.map(to_value)
+		})
+	}
+
+	/// Returns the code at given address at given time (block number).
+	fn code_at(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params_default_second::<RpcH160>(params).and_then(|(address, block_number,)| {
+			Eth::code_at(self, &address.into(), block_number)
+				.map(Bytes::from).map(to_value)
+		})
+	}
+
+	/// Sends signed transaction.
+	fn send_raw_transaction(&self, params: Params) -> Result<Value, Error> {
+		try!(self.active());
+
+		from_params::<(Bytes,)>(params).and_then(|(raw,)| {
+			let raw = raw.to_vec();
+			match UntrustedRlp::new(&raw).as_val() {
+				Ok(signed_transaction) => Eth::send_raw_transaction(self, signed_transaction)
+					.map(RpcH256::from).map(to_value),
+				Err(_) => Ok(to_value(RpcH256::from(H256::from(0)))),
```

Comment:
Agreed, this is just continuing the current behavior but an "invalid RLP" error would be nice.